src/kw_env.sh

```diff
@@ -213,42 +217,42 @@ function create_new_env()
   cmd_manager "$flag" "$cmd"
 
   # Check if the env name was not created
-  output=$(find "${local_kw_configs}/${ENV_DIR}" -type d -name "$env_name")
+  output=$(find "${local_kw_configs}/${ENV_DIR}" -type d -name "$encoded_env_name")
   if [[ -n "$output" ]]; then
     warning "It looks that you already have '$env_name' environment"
     warning 'Please, choose a new environment name.'
     return 22 # EINVAL
   fi
 
   # Create env folder
-  cmd="mkdir -p ${local_kw_configs}/${ENV_DIR}/${env_name}"
+  cmd="mkdir -p ${local_kw_configs}/${ENV_DIR}/${encoded_env_name}"
   cmd_manager "$flag" "$cmd"
 
   # Copy local configs
   for config in "${config_file_list[@]}"; do
     if [[ ! -e "${local_kw_configs}/${config}.config" ]]; then
       say "${config}.config does not exist. Creating a default one."
-      cmd="cp ${KW_ETC_DIR}/${config}.config ${local_kw_configs}/${ENV_DIR}/${env_name}"
+      cmd="cp ${KW_ETC_DIR}/${config}.config ${local_kw_configs}/${ENV_DIR}/${encoded_env_name}"
       cmd_manager "$flag" "$cmd"
     else
-      cmd="cp ${local_kw_configs}/${config}.config ${local_kw_configs}/${ENV_DIR}/${env_name}"
+      cmd="cp ${local_kw_configs}/${config}.config ${local_kw_configs}/${ENV_DIR}/${encoded_env_name}"
       cmd_manager "$flag" "$cmd"
     fi
   done
 
   # Handle build and config folder
-  cmd="mkdir -p ${cache_build_path}/${ENV_DIR}/${env_name}"
+  cmd="mkdir -p ${cache_build_path}/${ENV_DIR}/${encoded_env_name}"
   cmd_manager "$flag" "$cmd"
 
   current_env_name=$(get_current_env_name)
   ret="$?"
   # If we already have an env, we should copy the config file from it.
   if [[ "$ret" == 0 ]]; then
-    cmd="cp ${cache_build_path}/${ENV_DIR}/${current_env_name}/.config ${cache_build_path}/${ENV_DIR}/${env_name}/.config"
+    cmd="cp ${cache_build_path}/${ENV_DIR}/${current_env_name}/.config ${cache_build_path}/${ENV_DIR}/${encoded_env_name}/.config"
```

Comment:
As I commented before, it would be better if we just create a new folder with the encoded name (without the env name), and inside it we create the env name. For example, suppose the encoded_pwd is `845d87776ab0bcd0c17f051d264f1cf2`, and the env names are `RPI64` and `x86_64`, we would have something like this:

`${cache_build_path}/${ENV_DIR}/${encoded_pwd}/${current_env_name}/`